django/db/backends/mysql/operations.py

```diff
@@ -70,7 +70,7 @@ def date_trunc_sql(self, lookup_type, field_name):
 
     def _convert_field_to_tz(self, field_name, tzname):
         if settings.USE_TZ:
-            field_name = "CONVERT_TZ(%s, 'UTC', '%s')" % (field_name, tzname)
+            field_name = "CONVERT_TZ(%s, '%s', '%s')" % (field_name, self.connection.timezone_name, tzname)
```

Comment:
I think we should convert only when time-zones are different (like in original patch).

Reply:
Quick link: https://github.com/django/django/pull/8714/files